src/font/lv_font_fmt_txt.c

```diff
@@ -124,72 +125,75 @@ const void * lv_font_get_bitmap_fmt_txt(lv_font_glyph_dsc_t * g_dsc, lv_draw_buf
+                /*Handle stride*/
+                if(stride_in) {
+                    i = 0;  /*If there is a stride start from the next byte in the next line*/
+                    bitmap_in += line_rem;
                 }
-                bitmap_out_tmp += stride;
+                bitmap_out_tmp += stride_out;
             }
         }
         else if(fdsc->bpp == 2) {
             for(y = 0; y < gdsc->box_h; y ++) {
+                uint16_t line_rem = stride_in;
                 for(x = 0; x < gdsc->box_w; x++, i++) {
                     i = i & 0x3;
                     if(i == 0) bitmap_out_tmp[x] = opa2_table[(*bitmap_in) >> 6];
                     else if(i == 1) bitmap_out_tmp[x] = opa2_table[((*bitmap_in) >> 4) & 0x3];
                     else if(i == 2) bitmap_out_tmp[x] = opa2_table[((*bitmap_in) >> 2) & 0x3];
                     else if(i == 3) {
                         bitmap_out_tmp[x] = opa2_table[((*bitmap_in) >> 0) & 0x3];
+                        line_rem--;
                         bitmap_in++;
                     }
                 }
 
-                /*Go to the next byte if stopped in the middle of a byte and
-                 *the next line is byte aligned*/
-                if(byte_aligned && i != 0) {
-                    i = 0;
-                    bitmap_in++;
+                /*Handle stride*/
+                if(stride_in) {
+                    i = 0;  /*If there is a stride start from the next byte in the next line*/
+                    bitmap_in += line_rem;
                 }
-
-                bitmap_out_tmp += stride;
+                bitmap_out_tmp += stride_out;
             }
 
         }
         else if(fdsc->bpp == 4) {
             for(y = 0; y < gdsc->box_h; y ++) {
+                uint16_t line_rem = stride_in;
                 for(x = 0; x < gdsc->box_w; x++, i++) {
                     i = i & 0x1;
                     if(i == 0) {
                         bitmap_out_tmp[x] = opa4_table[(*bitmap_in) >> 4];
                     }
                     else if(i == 1) {
                         bitmap_out_tmp[x] = opa4_table[(*bitmap_in) & 0xF];
+                        line_rem--;
                         bitmap_in++;
                     }
                 }
 
-                /*Go to the next byte if stopped in the middle of a byte and
-                 *the next line is byte aligned*/
-                if(byte_aligned && i != 0) {
-                    i = 0;
-                    bitmap_in++;
+                /*Handle stride*/
+                if(stride_in) {
+                    i = 0;  /*If there is a stride start from the next byte in the next line*/
+                    bitmap_in += line_rem;
                 }
-
-                bitmap_out_tmp += stride;
+                bitmap_out_tmp += stride_out;
             }
         }
         else if(fdsc->bpp == 8) {
             for(y = 0; y < gdsc->box_h; y ++) {
+                uint16_t line_rem = stride_in;
```

Comment:
@kisvegabor after checking the failing test on CI, I've observed that in this case, when it's used with:
```c
uint16_t line_rem = stride_in;
```
If the stride is 0, then line_rem will go to the maximum value of the uint16_t, and it will overflow.

I suggest changing this to:

```c
uint16_t line_rem = stride_in != 0 ? stride_in : gdsc->box_w;
```

Reply:
Great catch! I agree with your proposal. Please add it for the other `bpp`s too.